x/feegrant/filtered_fee_test.go

```diff
@@ -0,0 +1,208 @@
+package feegrant_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/x/feegrant"
+	ocproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cosmos/cosmos-sdk/simapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func TestFilteredFeeValidAllow(t *testing.T) {
+	app := simapp.Setup(t, false)
+
+	ctx := app.BaseApp.NewContext(false, ocproto.Header{})
+	badTime := ctx.BlockTime().AddDate(0, 0, -1)
+	allowace := &feegrant.BasicAllowance{
+		Expiration: &badTime,
+	}
+	require.Error(t, allowace.ValidateBasic())
+
+	ctx = app.BaseApp.NewContext(false, ocproto.Header{
+		Time: time.Now(),
+	})
+	eth := sdk.NewCoins(sdk.NewInt64Coin("eth", 10))
+	atom := sdk.NewCoins(sdk.NewInt64Coin("atom", 555))
+	smallAtom := sdk.NewCoins(sdk.NewInt64Coin("atom", 43))
+	bigAtom := sdk.NewCoins(sdk.NewInt64Coin("atom", 1000))
+	leftAtom := sdk.NewCoins(sdk.NewInt64Coin("atom", 512))
+	now := ctx.BlockTime()
+	oneHour := now.Add(1 * time.Hour)
+
+	cases := map[string]struct {
+		allowance *feegrant.BasicAllowance
+		msgs      []string
+		// all other checks are ignored if valid=false
+		fee       sdk.Coins
+		blockTime time.Time
+		valid     bool
+		accept    bool
+		remove    bool
+		remains   sdk.Coins
+	}{
+		"msg contained": {
+			allowance: &feegrant.BasicAllowance{},
+			msgs:      []string{"/cosmos.feegrant.v1beta1.MsgRevokeAllowance"},
+			accept:    true,
+		},
```

Comment:
It is the simplest case. In the test, the called message would be `/cosmos.feegrant.v1beta1.MsgRevokeAllowance`, so it must be accepted.